src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -140,5 +205,11 @@ private static void GenerateSemanticBaseline(IEnumerable<int>? actual, string ba
 
             return results.ToArray();
         }
+
+        private class VSSemanticTokensResponse : SemanticTokens
```

Comment:
Interesting, why can't we use the language servers real VS semantic tokens response type that has isFinalized?

Reply:
In our semantic tokens logic today, we use a custom VS semantic tokens response type which isn't accessible in the server test layer since it lives in the client layer:
https://github.com/dotnet/razor-tooling/blob/main/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/VSSemanticTokensResponse.cs
So I thought a solution would be to just create a separate class here. But the class might be going away depending on the eventual fate of the IsFinalized bug 😓